packages/at_secondary_server/lib/src/verb/handler/pkam_verb_handler.dart

```diff
@@ -45,31 +46,19 @@ class PkamVerbHandler extends AbstractVerbHandler {
     var enrollId = verbParams[enrollmentId];
     var atSign = AtSecondaryServerImpl.getInstance().currentAtSign;
     var pkamAuthType = AuthType.pkamLegacy;
-    var publicKey;
+    String? publicKey;
 
     // Use APKAM public key for verification if enrollId is passed. Otherwise use legacy pkam public key.
     if (enrollId != null && enrollId.isNotEmpty) {
       var key =
           '$enrollId.$newEnrollmentKeyPattern.$enrollManageNamespace$atSign';
-      var enrollData = await keyStore.get(key);
-      if (enrollData != null) {
-        final atData = enrollData.data;
-        logger.finer('enrollData: $atData');
-        final enrollDataStoreValue =
-            EnrollDataStoreValue.fromJson(jsonDecode(atData));
-        if (enrollDataStoreValue.approval != null &&
-            enrollDataStoreValue.approval!.state ==
-                EnrollStatus.approved.name) {
-          publicKey =
-              EnrollDataStoreValue.fromJson(jsonDecode(atData)).apkamPublicKey;
-          pkamAuthType = AuthType.apkam;
-        } else {
-          response.data = 'failure';
-          response.isError = true;
-          response.errorMessage = 'enrollment id: $enrollId is not approved';
-          throw UnAuthenticatedException(response.errorMessage);
-        }
-      }
+      response = await handleEnrollment(key, enrollId, response);
+      // fetch publicKey from response
+      // _handleEnrollment store publicKey as response.data
+      publicKey = response.data;
+      pkamAuthType = AuthType.apkam;
+      // set response.data to success which will be sent as the verb response
+      response.data = 'success';
```

Comment:
At this point, the authentication is not completed. Only APKAM public key is fetched. So, suggest to avoid setting response.data to "success".
Also, suggest to rename "handleEnrollment" as "fetchAPKAMPublicKey" to clearly describe the method purpose.

Reply:
These changes have been made in [d6c8d9b](https://github.com/atsign-foundation/at_server/pull/1520/commits/d6c8d9b4373d495161dcb746ce1551715f1ca1d0)